lib/fastlane/plugin/wpmreleasetoolkit/versioning/files/ios_version_file.rb

```diff
@@ -0,0 +1,68 @@
+# A class for reading and writing version information to an Xcode .xcconfig file.
+require 'xcodeproj'
+
+module Fastlane
+  module Wpmreleasetoolkit
+    module Versioning
+      class IOSVersionFile
+        attr_reader :xcconfig_path
+
+        # Initializes a new instance of IOSVersionFile with the specified .xcconfig file path.
+        #
+        # @param [String] xcconfig_path The path to the .xcconfig file.
+        #
+        def initialize(xcconfig_path:)
+          @xcconfig_path = xcconfig_path
+        end
+
+        # Reads the app version from the .xcconfig file and returns it as a String.
+        #
+        # @return [String] The app version.
+        #
+        def read_app_version
+          verify_xcconfig_exists
+
+          config = Xcodeproj::Config.new(xcconfig_path)
+          config.attributes['VERSION_LONG']
+        end
+
+        # Reads the build code from the .xcconfig file and returns it String.
+        #
+        # @return [String] The build code.
+        #
+        def read_build_code
+          verify_xcconfig_exists
+
+          config = Xcodeproj::Config.new(xcconfig_path)
+          config.attributes['BUILD_NUMBER']
+        end
+
+        # Writes the provided version numbers to the .xcconfig file.
+        #
+        # @param [String] version_short The short version string.
+        # @param [String, nil] version_long The long version string (optional).
+        # @param [String, nil] build_number The build number (optional).
+        #
+        # version_short is optional because some apps (such as Day One iOS/Mac or Simplenote Mac) don't use it.
+        # build_number is optional because some apps (such as WP/JP iOS or WCiOS) don't use it.
+        #
+        def write(version_short:, version_long: nil, build_number: nil)
+          verify_xcconfig_exists
+
+          config = Xcodeproj::Config.new(xcconfig_path)
+          config.attributes['VERSION_SHORT'] = version_short.to_s unless version_short.nil?
+          config.attributes['VERSION_LONG'] = version_long.to_s unless version_long.nil?
+          config.attributes['BUILD_NUMBER'] = build_number.to_s unless build_number.nil?
+          config.save_as(Pathname.new(xcconfig_path))
+        end
+
+        # Verifies the existence of the .xcconfig file.
+        #
+        # @raise [UI.user_error] Raised if the .xcconfig file does not exist.
+        def verify_xcconfig_exists
```

Comment:
💄 Nitpick: same as above
```suggestion
        def verify_xcconfig_exists!
```

Reply:
I ended up moving the .exists? check into the initializer because these methods in these classes always take in the version file. It seemed to make more sense to do the validation there instead of having these calls throughout the class.

https://github.com/wordpress-mobile/release-toolkit/commit/5f77bed1c17e2a79ea50a3a9c0de48f4b7b5d8e9